libs/service-logging/src/service_logging/middleware.clj

```diff
@@ -0,0 +1,82 @@
+(ns service-logging.middleware
+  (:import [org.slf4j MDC])
+  (:use [slingshot.slingshot :only [try+ throw+]] )
+  (:require [clojure.string :as string]
+            [clojure.tools.logging :as log]
+            [clojure.data.json :as json]
+            [service-logging.thread-context :as tc]))
+
+(defn flatten-map
+  "Takes a nested map, and flattens it. The nested keys are combined with
+  the given separator, and the keys can optionally be pre-prended with a string"
+  [form separator pre]
+  (into {}
+        (mapcat (fn [[k v]]
+                  (let [prefix (if pre (str pre separator (name k)) (name k))]
+                    (if (map? v)
+                      (flatten-map v separator prefix)
+                      [[(keyword prefix) v]])))
+                form)))
+
+; TODO: Setup logger level conditional logging, which logs the body if the log
+;       level is at it's highest
+(defn- clean-request
+  [request]
+  (dissoc request
+          :body
+          :user-info))
```

Comment:
We should probably go ahead and cleanup compojure-api metadata for endpoints that are using it.
For example: https://github.com/iPlantCollaborativeOpenSource/DiscoveryEnvironmentBackend/blob/beta/services/metadata/src/metadata/util/service.clj#L30-L31
As far as I can tell, these `:compojure.api.middleware/options` and `:ring.swagger.middleware/data` keys contain the same JSON every time, for each endpoint, and one of them is huge since it's pretty much the entire swagger endpoint docs.
If an endpoint is not using compojure-api, then the `dissoc` for these keys is a no-op.